js/tools.js

```diff
@@ -0,0 +1,334 @@
+let selectButton = document.getElementById("select");
```

Comment:
I think these and some considerable bit below should be wrapped in an `onload`

Reply:
fixing this will require some amount of refactoring; creating a separate issue for this